tests/test_sshfsmount.cpp

```diff
@@ -93,43 +93,77 @@ struct SshfsMount : public mp::test::SftpServerTest
             {
-                // Check if the first of the remaining commands is being executed.
-                if (cmd == next_expected_cmd->first)
+                // Check if the first element of the given commands list is what we are expecting. In that case,
+                // give the correct answer. If not, check the rest of the list to see if we broke the execution
+                // order.
+                CommandVector::const_iterator found_cmd =
+                    std::find_if(next_expected_cmd, commands.end(), [&cmd](auto it) { return cmd == it.first; });
+
+                if (found_cmd == next_expected_cmd)
                 {
+                    invoked = true;
                     output = next_expected_cmd->second;
                     remaining = output.size();
                     ++next_expected_cmd;
-                    invoked = true;
 
                     return SSH_OK;
                 }
-                else
+                else if (found_cmd != commands.end())
                 {
-                    // If not, then we have to check the rest of the remaining
-                    // commands. If the executed command is there, it means that the
-                    // executed commands do not follow the order of our expected
-                    // command vector.
-                    for (auto it = std::next(next_expected_cmd); it != commands.end(); ++it)
-                    {
-                        if (cmd == it->first)
-                        {
-                            output = it->second;
-                            remaining = output.size();
-                            ADD_FAILURE() << "\"" << (it->first) << "\" executed out of order; expected \""
-                                          << next_expected_cmd->first << "\"";
-                            return SSH_OK;
-                        }
-                    }
+                    output = found_cmd->second;
+                    remaining = output.size();
+                    ADD_FAILURE() << "\"" << (found_cmd->first) << "\" executed out of order; expected \""
+                                  << next_expected_cmd->first << "\"";
+
+                    return SSH_OK;
                 }
             }
 
+            // If the command list was entirely checked or if the executed command is not on the list, check the
+            // default commands to see if there is an answer to the current command. If not, answer with a
+            // newline, because all strings returned by the server we are mocking end in newline.
             auto it = default_cmds.find(cmd);
             if (it != default_cmds.end())
             {
                 output = it->second;
                 remaining = output.size();
                 invoked = true;
             }
+            else
+            {
+                output = "\n";
+                remaining = output.size();
+                invoked = true;
+            }
```

Comment:
How about making `make_channel_read_return` always append / return a newline, rather than being explicit about it in all the return strings?

@townsend2010 could use your opinion here… I could go with either…

Reply:
That will certainly make the test a bit clearer in the sense that we won't have those `\n` in every answer. We can do it.
In case we need an answer without the final newline, we can just write a return function by hand.